factory/base.py

```diff
@@ -656,7 +656,7 @@ def __init__(self, **kwargs):
             setattr(self, field, value)
 
 
-class StubFactory(Factory):
+class StubFactory(Factory[T]):
```

Comment:
I believe this should be `StubFactory(Factory[StubObject])`, as we're returning a concrete class